### ISSUE TITLE
$set() does not update `KeyValue` Field the first time

### ISSUE THREAD

**Issue by Z3d0X** [filamentphp/filament#2939]
### Package

filament/forms

### Package Version

v2.13.10

### Laravel Version

v9.18.0

### Livewire Version

v2.10.5

### PHP Version

PHP 8.1

### Bug description

Using `$set` on a keyvalue field doesn't work the first time, but it works after that

### Steps to reproduce

In this example, the first state update to `children_no` doesn't fill `children_ages`, but it DOES fill on subsequent state updates

```php
TextInput::make('children_no')
    ->reactive()
    ->afterStateUpdated(function ($state, $set) {
        if (filled($state)) {
            $generated = array_fill_keys(range(1, $state), '');
            //$generated = [1 => '', 2 => '', ....]
            $set('children_ages', $generated);
        }
    })
    ->integer(),

KeyValue::make('children_ages')
    ->disableAddingRows()
    ->disableDeletingRows()
    ->disableEditingKeys(),
```

**Comment by alperenersoy** [comment]
I don't know if that's a good practice but I usually call afterStateUpdated in afterStateHydrated for these cases like:

```php
TextInput::make('children_no')
    ->reactive()
    ->afterStateUpdated(function ($state, $set) {
        if (filled($state)) {
            $generated = array_fill_keys(range(1, $state), '');
            //$generated = [1 => '', 2 => '', ....]
            $set('children_ages', $generated);
        }
    })
    ->afterStateHydrated(fn ($component) => $component->callAfterStateUpdated())
    ->integer(),
```

**Comment by zepfietje** [comment]
Can you provide us a reproduction repository URL and then we'll look at your issue?

That would allow us to download it and review your bug much easier, so it can be fixed quicker. Please make sure to give us some instructions on where to find the bug, and include a database seeder with everything we need to set it up quickly.

**Comment by Z3d0X** [comment]
@zepfietje  https://github.com/z3d0x/filament-demo/tree/issue/2939 here you go.

Its a fork of demo, but the only relevant file is `app/Filament/Pages/BugPage.php`.

Let me also summarize the issue and new findings.
the **_FIRST_** state update to `children_no` doesn't update `children_ages` UI, but it **_DOES_** on subsequent state updates.

I have also found out that **_FIRST_** state update to `children_no` does update the state of  `children_ages`, however it is not reflected in the UI

**Comment by danharrin** [comment]
Fixed by #3408.
